bot/exts/info/tags.py

```diff
@@ -172,123 +259,156 @@ async def search_tag_content(self, ctx: Context, *, keywords: str) -> None:
-                        "channel": ctx.channel.id
-                    }
-
-                self.bot.stats.incr(f"tags.usages.{tag['title'].replace('-', '_')}")
 
-                await wait_for_deletion(
-                    await ctx.send(embed=Embed.from_dict(tag['embed'])),
-                    [ctx.author.id],
-                )
+    async def list_all_tags(self, ctx: Context) -> None:
+        """Send a paginator with all loaded tags accessible by `ctx.author`, groups first, and alphabetically sorted."""
+        def tag_sort_key(tag_item: tuple[TagIdentifier, tag]) -> str:
+            group, name = tag_item[0]
+            if group is None:
+                # Max codepoint character to force tags without a group to the end
+                group = chr(0x10ffff)
+
+            return group + name
+
+        result_lines = []
+        current_group = object()
+        group_accessible = True
+
+        for identifier, tag in sorted(self.tags.items(), key=tag_sort_key):
+
+            if identifier.group != current_group:
+                if not group_accessible:
+                    # Remove group separator line if no tags in the previous group were accessible by the user.
+                    result_lines.pop()
+                # A new group began, add a separator with the group name.
+                current_group = identifier.group
+                if current_group is not None:
+                    group_accessible = False
+                    result_lines.append(f"\n\N{BULLET} **{current_group}**")
+                else:
+                    result_lines.append("\n\N{BULLET}")
+
+            if tag.accessible_by(ctx.author):
+                result_lines.append(f"**\N{RIGHT-POINTING DOUBLE ANGLE QUOTATION MARK}** {identifier.name}")
+                group_accessible = True
+
+        embed = Embed(title="Current tags")
+        await LinePaginator.paginate(result_lines, ctx, embed, max_lines=15, empty=False, footer_text=FOOTER_TEXT)
+
+    async def list_tags_in_group(self, ctx: Context, group: str) -> None:
+        """Send a paginator with all tags under `group`, that are accessible by `ctx.author`."""
+        embed = Embed(title=f"Tags under *{group}*")
+        tag_lines = sorted(
+            f"**\N{RIGHT-POINTING DOUBLE ANGLE QUOTATION MARK}** {identifier}"
+            for identifier, tag in self.tags.items()
+            if identifier.group == group and tag.accessible_by(ctx.author)
+        )
+        await LinePaginator.paginate(tag_lines, ctx, embed, footer_text=FOOTER_TEXT, empty=False, max_lines=15)
+
+    @tags_group.command(name="get", aliases=("show", "g"))
+    async def get_command(
+            self, ctx: Context,
+            tag_name_or_group: TagNameConverter = None,
+            tag_name: TagNameConverter = None,
+    ) -> bool:
+        """
+        When arguments are passed in:
+        If a single argument is given and it matches a group name, list accessible all tags from that group.
```

Comment:
I think this sentence needs revising.

Reply:
I'm open to any suggestions.
Is "If a single argument matching a group name is given, list accessible all tags from that group." better?

Reply:
Specifically I was first talking about "list accessible all tags", but I like what you suggested better than what is currently there. So "If a single argument matching a group name is given, list all accessible tags from that group." it is

Reply:
Oh I didn't even notice that, will fix